shared-data/python/opentrons_shared_data/pipette/load_data.py

```diff
@@ -0,0 +1,59 @@
+import json
+import os
+
+from typing import Dict, Any
+from functools import lru_cache
+
+from .. import load_shared_data, get_shared_data_root
+
+from . import types
+from .pipette_definition import PipetteConfigurations
+
+
+LoadedConfiguration = Dict[types.PipetteChannelType, Dict[types.PipetteModelType, Any]]
+
+
+def _build_configuration_dictionary(
+    rel_path: str, version: types.PipetteVersionType
+) -> LoadedConfiguration:
+    _dict: LoadedConfiguration = {}
+    for pipette_type in types.PipetteChannelType:
+        pipette_type_path = get_shared_data_root() / rel_path / pipette_type.value
+        _dict[pipette_type] = {}
+        for dir_name in os.scandir(pipette_type_path):
+            model_key = types.PipetteModelType.convert_from_model(dir_name.name)
+            _dict[pipette_type][model_key] = json.loads(
+                load_shared_data(
+                    f"{pipette_type_path}/{dir_name.name}/{version.major}.json"
+                )
+            )
+    return _dict
```

Comment:
Related to my other comment in this file, it seems like given the actual usage none of the directory walking is necessary.

```suggestion
def _get_configuration_dictionary(
    config_type: Union["general", "geometry", "liquid"],
    channels: types.PipetteChannelType,
    max_volume: types.PipetteModelType,
    version: types.PipetteVersionType,
) -> LoadedConfiguration:
    config_path = (
        get_shared_data_root()
        / "pipette"
        / "definitions"
        / "2"
        / config_type
        / channels.value
        / max_volume.value
        / f"{version.major}.json"
    )

    return json.loads(load_shared_data(config_path))
```